components/core/src/clp_s/clp-s.cpp

```diff
@@ -174,20 +179,40 @@ bool search_archive(
     }
 
     std::unique_ptr<OutputHandler> output_handler;
-    if (command_line_arguments.get_mongodb_enabled()) {
-        output_handler = std::make_unique<ResultsCacheOutputHandler>(
-                command_line_arguments.get_mongodb_uri(),
-                command_line_arguments.get_mongodb_collection(),
-                command_line_arguments.get_batch_size(),
-                command_line_arguments.get_max_num_results()
-        );
-    } else if (command_line_arguments.get_network_destination_enabled()) {
-        output_handler = std::make_unique<NetworkOutputHandler>(
-                command_line_arguments.get_host(),
-                command_line_arguments.get_port()
-        );
-    } else {
-        output_handler = std::make_unique<StandardOutputHandler>();
+    try {
+        switch (command_line_arguments.get_output_handler_type()) {
+            case CommandLineArguments::OutputHandlerType::Network:
+                output_handler = std::make_unique<NetworkOutputHandler>(
+                        command_line_arguments.get_network_dest_host(),
+                        std::to_string(command_line_arguments.get_network_dest_port())
+                );
+                break;
+            case CommandLineArguments::OutputHandlerType::Reducer:
+                if (command_line_arguments.do_count_results_aggregation()) {
+                    output_handler = std::make_unique<CountOutputHandler>(reducer_socket_fd);
```

Comment:
Do you think it's better to move `reducer::connect_to_reducer` here so that we don't need `reducer-socket-fd` parameter for `search_archive` (or move it to the constructor of `CountOutputHandler`)?

Reply:
I prefer having it before we do any IO (i.e. before we open the archive reader) so that in the case where a job has been cancelled the reducer can reject the connection before we hit disk.